apmpackage/apm/data_stream/app_logs/elasticsearch/ingest_pipeline/default.yml

```diff
@@ -9,3 +9,14 @@ processors:
       name: observer_ids
   - pipeline:
       name: remove_ecs_version
+  - geoip:
+      if: ctx.event?.category == 'device'
+      field: client.ip
+      target_field: client.geo
+      ignore_missing: true
+      database_file: GeoLite2-City.mmdb
+      on_failure:
+        - remove:
+            field: client.ip
```

Comment:
Do we have to remove the field on failure?

Reply:
Thanks for taking a look Carson!

Regarding the first item, I'm a bit confused because I'm taking a look at [this log](https://edge-lite-oblt.kb.us-west2.gcp.elastic-cloud.com/app/discover#/doc/apm_static_index_pattern_id/.ds-logs-apm.app.opbeans_android-default-2023.09.24-000012?id=nRUey4oBpy6-VGJ8ASZU) in `edge-lite` that's not an error, which has the `client.ip` attribute. That's the kind of log I'm trying to append geo attrs with these changes so it seems like we might be able to work with those?

> It will also be helpful to have a system test so that this doesn't break in the future.

I'm up for adding those kinds of tests, although I'm struggling a bit with finding similar tests (that check log's enhancements) in the [current tests](https://github.com/elastic/apm-server/blob/main/systemtest/ingest_test.go) for ingest processes, and without some similar tests to take as a base for mine, I'm not sure how to set things up on my own in order to do the logs validations. Is there any other test file I'm missing where I could find some ingest tests that verify attrs appended to logs? Otherwise, I'm also up to jump on a call to discuss what steps I can follow to set up these kind of tests for logs.

Reply:
I don't see why we should tbh, although it seems like the same is done in other places when using the `geoip` processor which made me think that there might be a reason for it, so I just left it to keep consistency with all other usages, but I'm ok to remove this part if it's not needed.

Reply:
I'm fine with using that predefined pipeline, however, doing so would append geo attrs to **all** logs, instead of only the ones with the attribute `event.category` set to `device`. I'm not sure is that could cause problems to the apm server.

Reply:
@LikeTheSalad you can execute the pipeline conditionally as well:

```
- pipeline: 
      if: ...
      name: client_geoip
```

Reply:
I've just discussed this with Silvia. We think that unless there's a specific reason, we'd prefer running the pipeline without the `if` check. Since we are already running client_geoip ingest pipeline for most of the other data streams, the fact that it is missing in this app_logs data stream appears to be only an oversight. Also, client.ip is only populated for swift / android otel agents AND `capture_personal_data: true`, so it should be fine to run client_geoip whenever client.ip exists.

Reply:
Cool! That makes things simpler, cheers!

Reply:
I finished creating the system test for this use case and I also applied the changes to use the existing `client_geoip` pipeline without conditions.

I found out a couple of things while I was working on the test:

* The `capture_personal_data` seems to be enabled by default (at least for the system tests), because when I was adding a new config inside `systemtest/apmservertest/config.go`, I printed some logs [in here](https://github.com/elastic/apm-server/blob/main/internal/beater/server.go#L195) where we're reading the config and realized that the `if` body was always getting executed no matter if I added the new config or not.
* There were 2 reasons why I wasn't able to see geo attributes locally: The first one is that the `agent.name` has to be either `android/java` or `iOS/swift` for the `client.ip` to be populated (as mentioned by @carsonip), and the second one was that the client.ip value received by the server during the tests was `127.0.0.1` which the server doesn't know how to translate into a location (which makes sense), so I had to override the local IP during testing by setting a real one using the `X-Forwarded-For` header so that the server could find a location from the client.ip attr.

Reply:
- You're right. capture_personal_data is default to true for both apm-server standalone and integration.
- Overriding X-Forwarded-For sounds good.